src/collection.js

```diff
@@ -787,7 +787,13 @@ export default class Collection {
       return Promise.resolve(payload);
     }
     return waterfall(this.hooks[hookName].map(hook => {
-      return record => hook(payload, this);
+      return record => {
+        const result = hook(payload, this);
+        if (!result || !result.hasOwnProperty("changes")) {
+          throw new Error(`Invalid return value for hook: ${result} has no 'changes' property`);
```

Comment:
This is likely to render as "Invalid return value for hook: [Object object] has no 'changes' property"